DNN Platform/Website/Providers/DataProviders/SqlDataProvider/09.04.02.SqlDataProvider

```diff
@@ -8,7 +8,126 @@
 /*****                                                  *****/
 /************************************************************/
 
+IF  EXISTS (SELECT * FROM sys.objects WHERE object_id = OBJECT_ID(N'{databaseOwner}{objectQualifier}SetCorePackageVersions') AND type in (N'P', N'PC'))
+DROP PROCEDURE {databaseOwner}{objectQualifier}SetCorePackageVersions
+GO
 
+SET ANSI_NULLS ON
+GO
+SET QUOTED_IDENTIFIER ON
+GO
+CREATE PROCEDURE {databaseOwner}{objectQualifier}SetCorePackageVersions
+AS
+DECLARE @Version VARCHAR(10)
+SET @Version = (SELECT TOP 1
+ CAST(v.Major AS VARCHAR) + '.'
+ + CAST(v.Minor AS VARCHAR) + '.'
+ + CAST(v.Build AS VARCHAR)
+FROM {databaseOwner}{objectQualifier}Version v
+ORDER BY v.Major DESC, v.Minor DESC, v.Build DESC);
+UPDATE {databaseOwner}{objectQualifier}Packages
+SET Version=@Version
+WHERE [Name] IN ('DotNetNuke.Authentication',
+'DotNetNuke.SearchResults',
+'DotNetNuke.Security',
+'DotNetNuke.ACTIONBUTTONSkinObject',
+'DotNetNuke.ACTIONSSkinObject',
+'DotNetNuke.BANNERSkinObject',
+'DotNetNuke.BREADCRUMBSkinObject',
+'DotNetNuke.COPYRIGHTSkinObject',
+'DotNetNuke.CURRENTDATESkinObject',
+'DotNetNuke.DOTNETNUKESkinObject',
+'DotNetNuke.DROPDOWNACTIONSSkinObject',
+'DotNetNuke.HELPSkinObject',
+'DotNetNuke.HOSTNAMESkinObject',
+'DotNetNuke.ICONSkinObject',
+'DotNetNuke.LANGUAGESkinObject',
+'DotNetNuke.LINKACTIONSSkinObject',
+'DotNetNuke.LINKSSkinObject',
+'DotNetNuke.LOGINSkinObject',
+'DotNetNuke.LOGOSkinObject',
+'DotNetNuke.MENUSkinObject',
+'DotNetNuke.NAVSkinObject',
+'DotNetNuke.PRINTMODULESkinObject',
+'DotNetNuke.PRIVACYSkinObject',
+'DotNetNuke.SEARCHSkinObject',
+'DotNetNuke.SIGNINSkinObject',
+'DotNetNuke.TERMSSkinObject',
+'DotNetNuke.TITLESkinObject',
+'DotNetNuke.TREEVIEWSkinObject',
+'DotNetNuke.USERSkinObject',
+'DotNetNuke.VISIBILITYSkinObject',
```

Comment:
This seems like a maintenance issue waiting to happen… Can we not use something like `WHERE IsSystem=1` for, at least, most of these?

Reply:
If I understand correctly, the packages listed here have no manifest, some others do have a minifest and actually get installed as an extension, so if we do WHERE IsSystem=1 we could hide upgrade failures on other extensions and make it hard to troubleshoot. Does that make sense?

Reply:
Yeah, so under what conditions does this list grow and we need to maintain it?  Does this only apply to "legacy" built-in components, and we'll never have a new one like this?

Reply:
They are not legacy yet, but could become, not sure how skins will work after the move to .Net Core... But I haven't seen a new skinobject in ages so I doubt this list would grow. This being said since they are not packages, it would not be dramatic if one is forgotton, the main issue with packages versioning is that that won't upgrade if there was a change between versions and the version of the package is the same version as already installed.

Reply:
Yes, this is prone to maintenance. But the alternative is to include SQL like in the 9.3.2 script for every single release which I find worse. Besides: I don't think we will continue doing things this way.

I did look for alternatives to find these automatically, but it was not possible without introducing some level of risk of false positives and fals negatives. IMO we should go with this until we are weening ourselves off webforms and rewriting these.

Reply:
:+1: works for me, thanks for the discussion